spec/avro/builder_logical_types_spec.rb

```diff
@@ -81,42 +75,130 @@ def self.with_logical_types
 
       it { is_expected.to be_json_eql(expected.to_json) }
 
-      with_logical_types do
-        it "sets the logical type on the field" do
-          expect(schema.fields.first.type.logical_type).to eq('timestamp-millis')
+      it "sets the logical type on the field" do
+        expect(schema.fields.first.type.logical_type).to eq('timestamp-millis')
+      end
+    end
+
+    context "decimal bytes logical type" do
+      subject(:schema_json) do
+        described_class.build do
+          record :with_bytes_decimal do
+            required :value, :bytes, logical_type: 'decimal', precision: 5, scale: 2
+          end
+        end
+      end
+
+      let(:expected) do
+        {
+          type: :record,
+          name: :with_bytes_decimal,
+          fields: [{ name: :value, type: { type: :bytes, logicalType: 'decimal', precision: 5, scale: 2 } }]
+        }
+      end
+
+      it { is_expected.to be_json_eql(expected.to_json) }
+
+      it "sets the logical type on the field" do
+        expect(schema.fields.first.type.logical_type).to eq('decimal')
+      end
+
+      context "validation" do
+        subject(:schema_json) do
+          described_class.build do
+            record :with_bytes_decimal do
+              required :value, :bytes, logical_type: 'decimal'
+            end
+          end
+        end
+
+        with_decimal_validation do
+          it "relies on Avro schema parse to validate" do
+            expect do
+              schema
+            end.to raise_error(Avro::SchemaParseError, 'Precision must be positive')
```

Comment:
Not a blocker for this PR but does the underlying Avro library provide enough context in these error messages for clients to track down the source of any validation problems?

Reply:
No, it still doesn't provide any context for the schema parse errors. We could consider implementing that in this library instead, validating each type as it serialized and rescuing any errors above to provide context.

The Avro library could obviously do something similar, though I'd argue that changing the errors is an API change and should probably wait for the next major release.